### ISSUE TITLE
Deprecate DeletionPolicy

### ISSUE THREAD

**Issue by lsviben** [crossplane/crossplane#5283]
### What problem are you facing?

DeletionPolicy was planned to be deprecated in [v1.12](https://github.com/crossplane/crossplane/releases/tag/v1.12.0) with the arrival of [ObserveOnly](https://github.com/crossplane/crossplane/blob/master/design/design-doc-observe-only-resources.md#deprecation-of-deletionpolicy) management policies, which then evaluated to [Granular management policies](https://github.com/crossplane/crossplane/blob/master/design/design-doc-observe-only-resources.md#deprecation-of-deletionpolicy) in [1.13](https://github.com/crossplane/crossplane/releases/tag/v1.13.0). In [1.14](https://github.com/crossplane/crossplane/releases/tag/v1.14.0) the Management Policies have graduated to BETA. 

So as management policies are on [road to GA](https://github.com/crossplane/crossplane/issues/4952), we should think about what to do with the deletionPolicy.
Do we want to remove it as management policies go to GA? 
Or leave it for a release after that? 
Do we plan to have some automated migration?
Are we still ok with management policies replacing deletion policy?

Hopefully this issue will answer those questions.

### How could Crossplane help solve your problem?

Find answers to the questions above, probably through a proposal?

**Comment by negz** [comment]
@lsviben Does DeletionPolicy log a deprecation warning at the moment? Are we pretty confident folks know it's deprecated?

**Comment by lsviben** [comment]
> @lsviben Does DeletionPolicy log a deprecation warning at the moment? Are we pretty confident folks know it's deprecated?

Yeah thats what I have been wondering myself, we were not really trying to get this point across, I i think I found why. 

```
// DeletionPolicy specifies what will happen to the underlying external
	// when this managed resource is deleted - either "Delete" or "Orphan" the
	// external resource.
	// This field is planned to be deprecated in favor of the ManagementPolicies
	// field in a future release. Currently, both could be set independently and
	// non-default values would be honored if the feature flag is enabled.
	// See the design doc for more information: https://github.com/crossplane/crossplane/blob/499895a25d1a1a0ba1604944ef98ac7a1a71f197/design/design-doc-observe-only-resources.md?plain=1#L223
	// +optional
	// +kubebuilder:default=Delete
```

We never actually deprecated it, just added a not that we plan to deprecate. So maybe for GA, the only thing to do is to deprecate it officially (if we are still ok with that)

**Comment by negz** [comment]
Yeah that sounds good to me as a first step.

**Comment by negz** [comment]
Whoops, automation wasn't supposed to close that.
